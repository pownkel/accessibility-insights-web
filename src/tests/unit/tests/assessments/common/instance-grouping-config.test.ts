// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import {
    getGroupByPropertyConfig,
    InstanceGroupingConfiguration,
} from 'assessments/common/instance-grouping-config';
import { GeneratedAssessmentInstance } from 'common/types/store-data/assessment-result-data';

describe('InstanceGroupingConfiguration', () => {
    describe.each([undefined, 'value'])('Property value = %s', propertyValue => {
        type PropertyBag = { property: string };

        const propertyName = 'property';
        const displayedPropertyName = 'Property';
        let instance: GeneratedAssessmentInstance<PropertyBag>;
        let groupingConfig: InstanceGroupingConfiguration;

        beforeEach(() => {
            groupingConfig = getGroupByPropertyConfig(propertyName, displayedPropertyName);
            instance = {
                propertyBag: { property: propertyValue },
            } as GeneratedAssessmentInstance<PropertyBag>;
        });

        it('getGroupKey', () => {
            const groupKey = groupingConfig.getGroupKey(instance);

            expect(groupKey).toMatchSnapshot();
        });

        it('getGroupTitle', () => {
            const groupTitle = groupingConfig.getGroupTitle(instance);

            expect(groupTitle).toMatchSnapshot();
        });
    });
});
