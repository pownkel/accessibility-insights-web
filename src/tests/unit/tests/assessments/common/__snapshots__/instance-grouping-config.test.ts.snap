// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstanceGroupingConfiguration Property value = undefined getGroupKey 1`] = `"(no value)"`;

exports[`InstanceGroupingConfiguration Property value = undefined getGroupTitle 1`] = `"Property = (no value)"`;

exports[`InstanceGroupingConfiguration Property value = value getGroupKey 1`] = `"value"`;

exports[`InstanceGroupingConfiguration Property value = value getGroupTitle 1`] = `"Property = value"`;
